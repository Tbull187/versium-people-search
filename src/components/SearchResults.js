import React from 'react';

const SearchResults = ({ searchResults, loading }) => {
  return (
    <div className='search-results'>
        {loading && <h2>Loading...</h2>}

        {searchResults.map(result => {
          return (
            <div className='result'>
              <div>First Name: {result.FirstName}</div>
              <div>Last Name: {result.LastName}</div>
              <div>City: {result.City}</div>
              <div>State: {result.State}</div>
            </div>
          )
        })}
    </div>
  )
}

export default SearchResults;