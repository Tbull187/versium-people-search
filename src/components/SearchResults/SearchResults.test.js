import React from 'react';
import SearchResults from './SearchResults';
import Spinner from 'react-bootstrap/Spinner';
import renderer from 'react-test-renderer';
import merge from 'lodash/merge';

describe('SearchResults Component', () => {
  it('should render results in a list', () => {
    const component = renderer.create(<SearchResults {...defaultProps} />).toJSON();
    expect(component).toMatchSnapshot();
  })

  it('should render a loading spinner', () => {
    const props = merge(defaultProps, {loading: true});
    const component = renderer.create(<SearchResults {...props}/>);

    const testInstance = component.root;
    expect(testInstance.findByType(Spinner))
  })
})

const defaultProps = {
  loading: false,
  searchResults: [
    {
      FirstName: 'Jon',
      LastName: 'Smith',
      Address: '123 Street',
      City: 'Seattle',
      State: 'WA'
    }
  ]
}