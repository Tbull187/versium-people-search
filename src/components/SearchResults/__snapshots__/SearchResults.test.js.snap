// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchResults Component should render results in a list 1`] = `
<div
  className="search-results"
>
  <div
    className="result"
  >
    <div
      className="bold"
    >
      Result #
      1
    </div>
    <div>
      First Name: 
      Jon
    </div>
    <div>
      Last Name: 
      Smith
    </div>
    <div>
      Address: 
      123 Street
    </div>
    <div>
      City: 
      Seattle
    </div>
    <div>
      State: 
      WA
    </div>
  </div>
</div>
`;

exports[`SearchResults Component should render search results in a list 1`] = `
<div
  className="search-results"
>
  <div
    className="result"
  >
    <div
      className="bold"
    >
      Result #
      1
    </div>
    <div>
      First Name: 
      Jon
    </div>
    <div>
      Last Name: 
      Smith
    </div>
    <div>
      Address: 
      123 Street
    </div>
    <div>
      City: 
      Seattle
    </div>
    <div>
      State: 
      WA
    </div>
  </div>
</div>
`;
